docs/src/main/asciidoc/spring-security.adoc

```diff
@@ -0,0 +1,252 @@
+             .statusCode(200)
+             .body(is("hello"));
+    }
+
+}
+----
+
+== Package and run the application
+
+Run the application with: `./mvnw compile quarkus:dev`.
+Open your browser to http://localhost:8080/greeting.
+
+The result should be: `{"message": "hello"}`.
+
+== Modify the controller to secure the `hello` method
+
+Since we are going to be using Spring Security annotations to define the authorizations to execute the `hello` method, you can add the Spring `@Secured` annotation allowing secure the resources by restricting which roles are able to execute it. The controller above will look as follows:
+
+[source,java]
+----
+package org.acme.spring.web;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.PathVariable;
+
+@RestController
+@RequestMapping("/greeting")
+public class GreetingController {
+
+    @Secured("admin")
+    @GetMapping
+    public String hello() {
+        return "hello";
+    }
+}
+----
+
+We use Quarkus `security-properties` extension to load the authentication information. It uses the main `application.properties` Quarkus configuration file to embed this information. Please, for more information about this extension check link:the assoiated guide[quarkus-elytron-security-properties-file].
+So, add the following content to `applications.properties` file:
+
+[source,properties]
+----
+quarkus.security.users.embedded.enabled=true
+quarkus.security.users.embedded.plain-text=true
+quarkus.security.users.embedded.users.scott=jb0ss
+quarkus.security.users.embedded.roles.scott=admin,user
+quarkus.security.users.embedded.users.stuart=test
+quarkus.security.users.embedded.roles.stuart=user
+----
+
+The corresponding test could look like:
+
+== GreetingControllerTest
+
+Note that a test for the controller has been created as well:
+
+[source, java]
+----
+package org.acme.spring.web;
+
+import io.quarkus.test.junit.QuarkusTest;
+import org.junit.jupiter.api.Test;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.is;
+
+@QuarkusTest
+public class GreetingControllerTest {
+
+    @Test
+    public void testHelloEndpointForbidden() {
+        given().auth().preemptive().basic("stuart", "test")
+                .when().get("/greeting")
+                .then()
+                .statusCode(403);
+    }
+
+    @Test
+    public void testHelloEndpoint() {
+        given().auth().preemptive().basic("scott", "jb0ss")
+                .when().get("/greeting")
+                .then()
+                .statusCode(200)
+                .body(is("hello"));
+    }
+
+}
+----
+== Test the changes
+
+Open your browser again to http://localhost:8080/greeting.
+
+The result should be:
+----
+Access to localhost was denied
+You don't have authorization to view this page.
+HTTP ERROR 403
+----
```

Comment:
Can we also add a working example with a valid auth? I think basic auth in the URL should work, no?